### PR TITLE
Support swupd indeterminate progress bar

### PR DESCRIPTION
**Pull request by jwakre**
Fixes Issue: #516

Changes proposed in this pull request:
- Create indeterminate progress bar when swupd StepCompletion == -1
- Add new swupd installation steps
- Add printPrefix to NewLooP()

### CODE REVIEW

**Comment by jwakre** [comment]
@mdhorn  @otaviobp This change must be coordinated with clearlinux/swupd-client#1142. The installer changes must be released first or at the same time to prevent unhandled installation steps which would appear as steps with no name for the installer.

File: swupd/swupd.go [hunk 1]
```diff
@@ -162,12 +162,20 @@ func (m Message) Process(printPrefix, line string) {
 			description = utils.Locale.Get("Resolving files that need to be installed")
 		case "download_packs":
 			description = utils.Locale.Get("Downloading required packs")
+		case "extract_packs":
+			description = utils.Locale.Get("Extracting required packs")
 		case "check_files_hash":
-			description = utils.Locale.Get("Verifying files integrity")
+			description = utils.Locale.Get("Verifying installed files")
```
Comment (jwakre) [inline swupd/swupd.go]: This description was updated to differentiate `check_files_hash` from `validate_fullfiles`. The `check_files_hash` step checks the hashes of files that are already installed on the target system. The `validate_fullfiles` step checks the hashes of fullfiles in the statedir that have not yet been installed to the target system.

**Comment by mdhorn** [comment]
My testing still shows at least two places during the installation process where it appears to "hang". The step show completed, but there is no new step that started for a long period of time.

The first is "Verifying files integrity"
IMG_1041
It shows completed, but then the installer sits "idle" for many seconds.

The step following is "Downloading missing files" does not show up for a while shown here:
IMG_1042

The second occurrence is "Installing base OS and configured bundles"
IMG_1043
It shows completed, but then the installer sits "idle" for many seconds.

The step following is "Installing boot loader" does not show up for a while shown here:
IMG_1044

I'm also attaching the clr-installer.log (as clr-installer.txt) from the run.
[clr-installer.txt](https://github.com/clearlinux/clr-installer/files/3730871/clr-installer.txt)

**Comment by jwakre** [comment]
@mdhorn Can you double check that you are using my PR with the swupd from clearlinux/swupd-client#1142?

A few things seem odd with your screenshots:

1. I changed the name of `Verifying files integrity` to `Verifying installed files`

2. You are missing the new steps that swupd added. Even if the installer did not correctly handle these new installation steps, there would be an installation step entry with a blank name.

**Comment by mdhorn** [comment]
> @mdhorn Can you double check that you are using my PR with the swupd from [clearlinux/swupd-client#1142](https://github.com/clearlinux/swupd-client/pull/1142)?
> 
> A few things seem odd with your screenshots:
> 
> 1. I changed the name of `Verifying files integrity` to `Verifying installed files`
> 2. You are missing the new steps that swupd added. Even if the installer did not correctly handle these new installation steps, there would be an installation step entry with a blank name.

:anguished:
Nope, you're right the last build was done against 583, not 582. Rebuilding and testing now.

**Comment by mdhorn** [comment]
> > @mdhorn Can you double check that you are using my PR with the swupd from [clearlinux/swupd-client#1142](https://github.com/clearlinux/swupd-client/pull/1142)?
> > A few things seem odd with your screenshots:
> > 
> > 1. I changed the name of `Verifying files integrity` to `Verifying installed files`
> > 2. You are missing the new steps that swupd added. Even if the installer did not correctly handle these new installation steps, there would be an installation step entry with a blank name.
> 
> 😧
> Nope, you're right the last build was done against 583, not 582. Rebuilding and testing now.

Rebuilding with the latest bits still shows pausing in the same two spots.
Here is the log file from the latest run.
[clr-installer_582.txt](https://github.com/clearlinux/clr-installer/files/3731592/clr-installer_582.txt)

I verified the sha on the build matches the head of the 582 branch.

**Comment by mdhorn** [comment]
> > > @mdhorn Can you double check that you are using my PR with the swupd from [clearlinux/swupd-client#1142](https://github.com/clearlinux/swupd-client/pull/1142)?
> > > A few things seem odd with your screenshots:
> > > 
> > > 1. I changed the name of `Verifying files integrity` to `Verifying installed files`
> > > 2. You are missing the new steps that swupd added. Even if the installer did not correctly handle these new installation steps, there would be an installation step entry with a blank name.
> > 
> > 
> > 😧
> > Nope, you're right the last build was done against 583, not 582. Rebuilding and testing now.
> 
> Rebuilding with the latest bits still shows pausing in the same two spots.
> Here is the log file from the latest run.
> [clr-installer_582.txt](https://github.com/clearlinux/clr-installer/files/3731592/clr-installer_582.txt)
> 
> I verified the sha on the build matches the head of the 582 branch.

Thanks for the help debugging this John.

The problem was the GUI not observing the /usr/local/bin path of root's path when launch from Gnome desktop.
See https://github.com/clearlinux/clr-installer/issues/586

**Comment by jwakre** [comment]
@mdhorn @otaviobp The installation steps that swupd will provide the installer are stable now, so I will merge the installer changes. Before I merge, I am going to verify that this PR works with the latest changes on clearlinux/swupd-client#1142.

**Pull request merged**